### PR TITLE
MS-205 Sync schedule revamp

### PR DESCRIPTION
**Pull request by luhmirin-s**
Main changes:
* Changed the `:infra:config-sync` module into `:infra:sync` because some aspects of the configuration sync already affected other sync workers, namely rescheduling of image upload with new constraints on project config sync and starting enrolment record upload on device config sync. It turned out pretty clear in the end.
* Moved several basic workers into the new sync module - image upload, enrollment record upload, and firmware download. All three workers have trivial implementation; this allows to either remove work manager dependency from the main infra module or remove the module altogether, like in the case of enrolment record sync.
* Everything related to scheduling background work is consolidated within `SyncOrchestrator`. In the case of event sync, this only affects the initial "master" worker, the remainder of event sync logic remains in the `:infra:event-sync` module as it is significantly more complex than other workers.
* Removed the last couple of sync-related pieces of code from the app module.

### CODE REVIEW

File: infra/sync/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            buildConfigField("long", "DEVICE_DOWN_SYNC_WORKER_INTERVAL_MINUTES", "30L")
+            buildConfigField("long", "IMAGE_UP_SYNC_WORKER_INTERVAL_MINUTES", "60L")
+            buildConfigField("long", "FIRMWARE_UPDATE_WORKER_INTERVAL_MINUTES", "1440L")
+            buildConfigField("long", "EVENT_SYNC_WORKER_INTERVAL_MINUTES", "60L")
+        }
+        getByName("staging") {
+            buildConfigField("long", "PROJECT_DOWN_SYNC_WORKER_INTERVAL_MINUTES", "15L")
+            buildConfigField("long", "DEVICE_DOWN_SYNC_WORKER_INTERVAL_MINUTES", "15L")
+            buildConfigField("long", "IMAGE_UP_SYNC_WORKER_INTERVAL_MINUTES", "15LL")
+            buildConfigField("long", "FIRMWARE_UPDATE_WORKER_INTERVAL_MINUTES", "15LL")
```
Comment (alexandr-simprints) [inline infra/sync/build.gradle.kts:21]: `15LL` - is this a typo in `IMAGE_UP_SYNC_WORKER_INTERVAL_MINUTES`, `EVENT_SYNC_WORKER_INTERVAL_MINUTES` and `FIRMWARE_UPDATE_WORKER_INTERVAL_MINUTES`?

**Review by alexandr-simprints — COMMENTED** [review]
LGTM. Approving once confirmed that these changes work as expected in builds with proguard enabled.

File: infra/sync/src/test/java/com/simprints/infra/sync/enrolments/EnrolmentRecordWorkerTest.kt [hunk 2]
```diff
@@ -24,11 +27,11 @@ class EnrolmentRecordWorkerTest {
...
     private val params = mockk<WorkerParameters>(relaxed = true) {
         every { inputData } returns workDataOf(
-            EnrolmentRecordSchedulerImpl.INPUT_ID_NAME to INSTRUCTION_ID,
-            EnrolmentRecordSchedulerImpl.INPUT_SUBJECT_IDS_NAME to arrayOf(SUBJECT_ID),
+            SyncConstants.RECORD_UPLOAD_INPUT_ID_NAME to INSTRUCTION_ID,
+            SyncConstants.RECORD_UPLOAD_INPUT_SUBJECT_IDS_NAME to arrayOf(SUBJECT_ID),
         )
     }
-    private val worker = EnrolmentRecordWorker(
+    private val worker = com.simprints.infra.sync.enrolments.EnrolmentRecordWorker(
```
Comment (meladRaouf) [inline infra/sync/src/test/java/com/simprints/infra/sync/enrolments/EnrolmentRecordWorkerTest.kt:34]: you can remove the com.simprints.infra.sync.enrolments.

File: infra/images/src/main/java/com/simprints/infra/images/ImageUpSyncScheduler.kt [hunk 3]
```diff
@@ -7,10 +7,7 @@ interface ImageUpSyncScheduler {
      */
```
Comment (meladRaouf) [inline infra/images/src/main/java/com/simprints/infra/images/ImageUpSyncScheduler.kt:7]: I think this file can be deleted

**Review by meladRaouf — APPROVED** [review]

**Pull request merged**